File: Arrays/One Dimensional Arrays/PalidromeString.py
arr = ['a', 'b', 'b', 'a']

def isPalindrome(arr):

    i = 0
    j = len(arr) - 1

    while (i < j):
        if (arr[i] == arr[j]):
            i += 1
            j -= 1
        else:
            return False
    return True

if (isPalindrome(arr)):
    print("Palindrome")
else:
    print("Not a Palindrome")

File: Arrays/One Dimensional Arrays/PalidromeString.cpp
#include <bits/stdc++.h>
using namespace std;

bool isPalindrome(char arr[]){
    int i = 0;
    int j = strlen(arr) - 1;

    while (i < j){
        if(arr[i] == arr[j]){
            i++;
            j--;
        }else{
            return false;
        }
        
    }
    return true;
}



int main(){
     
     char arr[100];
     cin.getline(arr, 100);

     if (isPalindrome(arr)){
         cout << "Palindromic String";
     }else{
         cout << "Not a Palindrome";
     }

    return 0;
}
